WordPress/gutenberg
# Add color and typography presets to Global Styles

## What?
A popular feature in classic themes was to allow users to change the color palette and font families of their theme, usually the customizer. This is already possible in Global Styles, but the UI to do so is tricky to use - it takes a lot of clicks and a deep understanding of how Global Styles works. This PR abstracts that complexity away for users so they can change colors and typography details easily without knowing how Global Styles works.

These color and typography settings are extracted from the style variations, and exposed in the Global Styles UI. 

## Why?
This makes it possible to mix and match color and typography settings from different style variations. In the future this will also allow theme developers to offer a much wider range of color and typography variations and potentially even mix and match these variations between themes.

## How?
- Process style variations to extract the color or typography rules from them. Note: This extracts all of the typography/color rules, not just those under "settings", but all of the rules in the style variation.
- Add a new UI to the colors and typography sections of the Global Styles UI to give access to these extracted settings

## Testing Instructions
1. Switch to a theme that has multiple style variations
2. Open the Site Editor
3. Go to the Global Styles panel
4. Open the Colors section
5. Open the Palette section
6. You should see a section called "Presets"
7. Select one of the presets
8. The colors of your preview should be updated to match the preset
9. Go back to the root of Global Styles
10. Open the Typography section
11. Select "Typesets"
12. Select one of the Preset typography options
13. The typography of the preview should update

## Screenshots or screencast 
<img width="280" alt="Screenshot 2023-12-07 at 17 07 08" src="https://github.com/WordPress/gutenberg/assets/275961/86a1c814-f860-4d3a-bb70-ad94464262b3">
<img width="280" alt="Screenshot 2023-12-07 at 16 48 04" src="https://github.com/WordPress/gutenberg/assets/275961/ca559b7c-efe4-4c58-9a8d-2442880dc033">
<img width="281" alt="Screenshot 2023-12-07 at 16 47 52" src="https://github.com/WordPress/gutenberg/assets/275961/9c3c0086-334e-4d32-aae4-3c0952a9cfe9">

There's also a demo video here: https://ben.blog/2023/12/07/global-styles-add-color-and-typography-presets/
## Todo
- Show the defaults for both color and typography
- Also apply the duotone from the style variation

## Questions
- Should we name each color/typography variation? What should they be called?
- Does the "Typesets" section in "Typography" make sense?
- How could we surface this in Site View?


## Notes
One of the powerful things about the approach taken here is that it will work for existing themes - it doesn't require theme developers to add special settings to their themes.
